java/arcs/sdk/Particle.kt

```diff
@@ -64,4 +64,18 @@ interface Particle {
         slotName: String,
         model: Map<String, Any> = mapOf()
     ): Map<String, Any>? = model
+
+
+    /**
+     * Register a reaction to an event (optional).
+     *
+     * Particle templates may emit events, usually from user actions.
+     *
+     * Only implemented for wasm, no-op on JVM.
+     *
+     * @param name The name of the triggered event
+     * @param handler A callback (consumer) in reaction to the event
+     * @see [renderOutput]
+     */
+    fun eventHandler(name: String, handler: (Map<String, String>) -> Unit) = Unit
```

Comment:
Why do we need this, is it just here as it was convenient to include or is it needed to fix IDE support somehow?

Reply:
We need this because adding IDE support revealed a compilation failure.

https://github.com/PolymerLabs/arcs/blob/a7fd8367bb6a84fbb9dfa86887fbe8483b6df19e/particles/Tutorial/Kotlin/4_Handles/GetPerson.kt#L23

Some particle code had a Wasm-only implementation. Since we're also compiling for the Jvm, I needed to make this method compatible by  moving it to a higher level.